runtime-extension-model/src/test/java/org/mule/runtime/core/api/extension/provider/MuleExtensionModelDeclarerTestCase.java

```diff
@@ -0,0 +1,103 @@
+  private final String isParseTemplateUseLegacyDefaultTargetValue;
+
+  public MuleExtensionModelDeclarerTestCase(String isDoNotSupportExpressions, String expectedResult,
+                                            String isParseTemplateUseLegacyDefaultTargetValue) {
+    this.isDoNotSupportExpressionsValue = isDoNotSupportExpressions;
+    this.expectedResult = expectedResult;
+    this.isParseTemplateUseLegacyDefaultTargetValue = isParseTemplateUseLegacyDefaultTargetValue;
+  }
+
+  // TODO W-13974116: Remove circular dependency when adding org.mule.tests:mule-tests-unit to this module
+  @Before
+  public void setUp() {
+    isDoNotSupportExpressionsOriginalValue =
+        setProperty(REVERT_SUPPORT_EXPRESSIONS_IN_VARIABLE_NAME_IN_SET_VARIABLE_PROPERTY, isDoNotSupportExpressionsValue);
+    isParseTemplateUseLegacyDefaultTargetValueOriginalValue =
+        setProperty(PARSE_TEMPLATE_USE_LEGACY_DEFAULT_TARGET_VALUE, isParseTemplateUseLegacyDefaultTargetValue);
+  }
+
+  @After
+  public void tearDown() {
+    if (isDoNotSupportExpressionsOriginalValue == null) {
+      clearProperty(REVERT_SUPPORT_EXPRESSIONS_IN_VARIABLE_NAME_IN_SET_VARIABLE_PROPERTY);
+    } else {
+      setProperty(REVERT_SUPPORT_EXPRESSIONS_IN_VARIABLE_NAME_IN_SET_VARIABLE_PROPERTY, isDoNotSupportExpressionsOriginalValue);
+
+    }
+    if (isParseTemplateUseLegacyDefaultTargetValueOriginalValue == null) {
+      clearProperty(PARSE_TEMPLATE_USE_LEGACY_DEFAULT_TARGET_VALUE);
+    } else {
+      setProperty(PARSE_TEMPLATE_USE_LEGACY_DEFAULT_TARGET_VALUE, isDoNotSupportExpressionsOriginalValue);
+    }
+  }
+
+  @Test
+  @Issue("W-13965819")
+  public void whenIsParseTemplateUseLegacyDefaultTargetValueTheTargetValueIsMessage() {
+    MuleExtensionModelDeclarer muleExtensionModelDeclarer =
+        new MuleExtensionModelDeclarer(mock(ComponentMetadataConfigurerFactory.class));
+    ExtensionDeclarer extensionDeclarer = muleExtensionModelDeclarer.createExtensionModel();
+    OperationDeclaration parseTemplateOperation = extensionDeclarer.getDeclaration().getOperations().stream()
+        .filter(operationDeclaration -> operationDeclaration.getName().equals("parseTemplate")).findFirst().get();
+    assertTargetValueParameter(parseTemplateOperation);
+  }
+
+  private void assertTargetValueParameter(OperationDeclaration parseTemplateOperation) {
+    if (parseBoolean(isParseTemplateUseLegacyDefaultTargetValue)) {
+      assertThat(parseTemplateOperation.getParameterGroup(ParameterGroupModel.OUTPUT).getParameters().stream()
+          .filter(parameterDeclaration -> parameterDeclaration.getName().equals("targetValue")).findFirst().get()
+          .getDefaultValue(),
+                 equalTo("#[message]"));
+    } else {
+      assertThat(parseTemplateOperation.getParameterGroup(ParameterGroupModel.OUTPUT).getParameters().stream()
+          .filter(parameterDeclaration -> parameterDeclaration.getName().equals("targetValue")).collect(toList()), is(empty()));
```

Comment:
Shouldn't this parameter actually exist regardless of the system property? Isn't the property only supposed to control the default value?

Reply:
I found the answer for this one myself while refactoring the test. In fact, the parameter is added to the ExtensionModel by the enricher if the Declarer does not define it. So if we assert over the declarer we will see that the parameter does not exist in the non-property version, but if we assert over the extension model, we will see that it does exist in both versions, only the default changes.